### DISCUSSION TITLE
Conditionally execute fallback (e.g. for specific mode)

### DISCUSSION THREAD

**Discussion by ovk** [saghen/blink.cmp#833] [Q&A] [answered]
### Feature Description

I've been trying to configure `blink.cmp` in such a way that `default` preset is used everywhere, but in command line I want `<Tab>` to be used to show completion options (instead of Ctrl+Space). While this can be done with a function for `<Tab>` mapping, I couldn't find and easy way to execute fallback when not in command line mode.

Here's example config:

```lua
{
    'saghen/blink.cmp',

    version = 'v0.8.2',

    opts = {
        keymap = {
            preset = 'default',

            ['<CR>'] = { 'accept', 'fallback' },

            ['<Tab>'] = {
                function(cmp)
                    if (vim.api.nvim_get_mode().mode == 'c') then
                        cmp.show()
                    else
                        -- Fallback ?
                    end
                end,
            },
        },

        completion = {
            menu = { auto_show = function(ctx) return ctx.mode ~= 'cmdline' end }
        },
    },
    opts_extend = { "sources.default" }
}
```

Without conditional fallback, only two options are available:

- If `fallback` is added to the list for `<Tab>` then command line completion doesn't work properly (`^I` inserted when Tab key pressed). 
- If `fallback` is not added, then Tab key stops working in insert mode.

**Comment by saghen** [comment, marked as answer]
Not sure I'm understanding correctly, but you can run the fallback by returning a non truthy value from the previous command, see: https://cmp.saghen.dev/configuration/keymap.html#example

```lua
['<C-n>'] = { 
  function(cmp)
    if (vim.api.nvim_get_mode().mode == 'c') then
      return cmp.show() -- returns true if it shows, otherwise nil
    end
    -- runs next command because we return nil
  end,
  "fallback"
}
```

**Reply by ovk** [reply]
Thank you, @Saghen. I didn't realize you can signal whether fallback should execute by returning a value from the previous command. I just gave it a try and it is exactly what I was looking for.
